src/pretty.rs

```diff
@@ -0,0 +1,471 @@
+                                    .append(t.to_owned().pretty(allocator))
+                                    .append(allocator.text(","))
+                            }),
+                            allocator.line(),
+                        )
+                        .append(allocator.line())
+                        .append(allocator.text("_ => "))
+                        .append(def.to_owned().pretty(allocator))
+                        .nest(2)
+                        .append(allocator.line_())
+                        .braces()
+                        .group(),
+                )
+                .append(allocator.space())
+                .append(tst.to_owned().pretty(allocator)),
+
+            Array(fields) => allocator
+                .line()
+                .append(allocator.intersperse(
+                    fields.iter().map(|rt| rt.to_owned().pretty(allocator)),
+                    allocator.text(",").append(allocator.line()),
+                ))
+                .nest(2)
+                .append(allocator.line())
+                .group()
+                .brackets(),
+
+            Op1(op, rt) => match op.pos() {
+                crate::term::OpPos::Infix => allocator.as_string(op).append(allocator.atom(rt)),
+                crate::term::OpPos::Postfix => allocator.atom(rt).append(allocator.as_string(op)),
+                crate::term::OpPos::Special => {
+                    use crate::term::UnaryOp::*;
+                    match op {
+                        Ite() => allocator
+                            .text("if ")
+                            .append(rt.to_owned().pretty(allocator)),
+                        op => panic!("pretty print is not impleented for {:?}", op),
+                    }
+                }
+            },
+            Op2(op, rtl, rtr) => allocator
+                .atom(rtl)
+                .append(allocator.space())
+                .append(op.clone().pretty(allocator))
+                .append(allocator.line())
+                .append(allocator.atom(rtr))
+                .nest(2)
+                .group(),
+            OpN(op, rts) => allocator
+                .as_string(op)
+                .append(allocator.line())
+                .append(
+                    allocator
+                        .intersperse(rts.iter().map(|rt| allocator.atom(rt)), allocator.line())
+                        .nest(2),
+                )
+                .group(),
+
+            Sym(sym) => allocator.text(format!("<symbol: {}>", sym)),
+
+            // TODO
+            Wrapped(i32, RichTerm) => unimplemented!(),
+
+            MetaValue(mv) => mv.to_owned().pretty(allocator),
+            Import(f) => allocator
+                .text("import ")
+                .append(allocator.as_string(f.to_string_lossy()).double_quotes()),
+            ResolvedImport(id) => allocator.text(format!("import <file_id: {:?}>", id)),
+            ParseError => allocator.text("## PARCE ERROR! ##"),
```

Comment:
Let's be consistent and use the same `<`/`>` for special constructs.

```suggestion
            ParseError => allocator.text("<parse error>"),
```